### PR TITLE
[FLINK-1753] Extend KafkaITCase, reworked PersistentKafkaSource (initially)

### PR DESCRIPTION
**Pull request by rmetzger**
Added large messages test and reworked PersistentKafkaSource.

A Flink user I'm talking with offline is affected by this issue, that's why I would like to merge this rather soon. (I am aware that one test case is failing, so I'll spend more time on this tomorrow morning)

### CODE REVIEW

**Comment by rmetzger** [comment]
The failed test has been resolved.

File: flink-staging/flink-streaming/flink-streaming-connectors/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaITCase.java [hunk 1]
```diff
@@ -709,4 +834,72 @@ private static KafkaServer getKafkaServer(int brokerId, File tmpFolder) throws U
...
+			p.send(new KeyedMessage<String, String>("testtopic", "What a message"));
+		}
+
+		LOG.info("+++++++++++ FINISHED PRODUCING ++++++++++++++++ ");
+
+		try {
+			Thread.sleep(1000000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
```
Comment (mbalassi) [inline flink-staging/flink-streaming/flink-streaming-connectors/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaITCase.java]: Why do you have this as comment?

**Comment by mbalassi** [comment]
LGTM besides that one question.

File: flink-staging/flink-streaming/flink-streaming-connectors/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaITCase.java (hunk 1 above)
Comment (rmetzger) [inline flink-staging/flink-streaming/flink-streaming-connectors/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaITCase.java]: I was so desperate that I wanted to test the Kafka stuff without any Flink code. But in the end I figured it out.

I can remove the code.

**Pull request closed without merging**